src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1601,53 +1614,64 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                 SCROLLLOCK_ON |
                 CAPSLOCK_ON;
 
-            if (keyDown && (modifiers.Value() & ~permittedModifiers) == 0)
+            if ((modifiers.Value() & ~permittedModifiers) == 0)
             {
                 auto& s = _altNumpadState;
 
                 if (vkey == VK_ADD)
                 {
-                    // Alt '+' <number> is used to input Unicode code points.
-                    // Every time you press + it resets the entire state
-                    // in the original OS implementation as well.
-                    s.encoding = AltNumpadEncoding::Unicode;
-                    s.accumulator = 0;
-                    s.active = true;
+                    if (keyDown)

```

Comment:
I had to move keyDown into these places so we could stash the keyUp event in case composition was active.

This makes `ALTdown KP+down KP+up ALTup` work.

Reply:
This may be easier to read if the `if (keyDown)` was outside the if/else chain instead of inside it.